src/cargo/sources/registry/mod.rs

```diff
@@ -801,14 +801,9 @@ impl<'cfg> Source for RegistrySource<'cfg> {
         self.yanked_whitelist.extend(pkgs);
     }
 
-    fn is_yanked(&mut self, pkg: PackageId) -> CargoResult<bool> {
+    fn is_yanked(&mut self, pkg: PackageId) -> Poll<CargoResult<bool>> {
         self.invalidate_cache();
```

Comment:
What is the impact of having the cache invalidated on every iteration of this loop?

Reply:
It's probably better to move the invalidation outside the loop. I'll add it before all existing callers.